sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_client_async.py

```diff
@@ -215,7 +216,29 @@ async def _do_retryable_operation_async(self, operation, *args, **kwargs):
                         break
                     await asyncio.sleep(self._retry_policy.get_backoff_time(retry_settings, exc))
                     if exc.condition == ErrorCondition.LinkDetachForced:
+                        total_link_credit = self._link.total_link_credit
                         await self._close_link_async()  # if link level error, close and open a new link
+                        if isinstance(self._link, SenderLink):
+                            self._link = self._session.create_sender_link(
+                                target_address=self.target,
+                                link_credit=self._link_credit,
```

Comment:
if we are storing total link credit from before the detach happened, should we pass in `total_link_credit` as `link_credit` here when recreating the links? Or do we let flow() get called to update the link credit that way?